### PR TITLE
crimson/osd: add_recovery(oid) before recover_object(oid)

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @liu-chunmei

File: src/crimson/osd/pg_recovery.cc [hunk 1]
```diff
@@ -258,10 +258,10 @@ std::optional<crimson::osd::blocking_future<>> PGRecovery::recover_missing(
   const hobject_t &soid, eversion_t need)
 {
   if (pg->get_peering_state().get_missing_loc().is_deleted(soid)) {
-    return pg->get_recovery_backend()->get_recovering(soid).make_blocking_future(
+    return pg->get_recovery_backend()->add_recovering(soid).make_blocking_future(
```
Comment (xxhdx1985126) [inline src/crimson/osd/pg_recovery.cc:261]: Maybe this line can be moved out of the "if (pg->get_peering_state().get_missing_loc().is_deleted(soid))" block? :-)

**Comment by xxhdx1985126** [comment]
LGTM:-)

**Review by xxhdx1985126 — COMMENTED** [review]
LGTM:-)

**Review by xxhdx1985126 — APPROVED** [review]

**Review by liu-chunmei — APPROVED** [review]
verified.   crimson cluster with osd >=3 startup.

**Pull request merged**